### PR TITLE
Poison to Jason replacement & Postgrex update changes

### PR DESCRIPTION
**Pull request by vasspilka**

### CODE REVIEW

File: test/support/json_serializer.ex [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
-        nil -> []
-        type -> type |> String.to_existing_atom() |> struct
+      case type do
+        [] -> Jason.decode!(binary)
+        type -> struct(type, to_atom_keys(Jason.decode!(binary)))
       end
+    end
 
-    Poison.decode!(binary, as: type)
+    defp to_atom_keys(map) do
```
[the remark this answers is not in the document]
Comment (vasspilka) [reply test/support/json_serializer.ex]: Thanks, will get it fixed

[the remark this answers is not in the document]
Comment (vasspilka) [reply test/support/json_serializer.ex]: Is fixed now

File: lib/event_store/json_serializer.ex [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def deserialize(binary, config) do
+      type =
+        case Keyword.get(config, :type, nil) do
+          nil -> []
+          type -> type |> String.to_existing_atom() |> struct
+        end
+
+      case type do
+        [] -> Jason.decode!(binary)
+        type -> struct(type, Jason.decode!(binary, keys: :atoms))
```
[the remark this answers is not in the document]
Comment (michalmuskala) [reply lib/event_store/json_serializer.ex]: Oh, the additional `struct` call is already there. So only the `:atoms` to `:atoms!` change would be nice.

[the remark this answers is not in the document]
Comment (vasspilka) [reply lib/event_store/json_serializer.ex]: On it

**Comment by vasspilka** [comment]
@danhawkins Hey, I am having trouble figuring out how to make the tests pass. If you manage to figure out why they don't we can fix and merge this.

**Comment by vasspilka** [comment]
I am not "actively" working on it, but I can try to have a look again these days. Also you started using commanded on Quiqup?

**Pull request merged**